subsys/bluetooth/services/fast_pair/adv_manager/locator_tag.c

```diff
@@ -0,0 +1,227 @@
+{
+	bt_fast_pair_adv_manager_request(
+		&bt_fast_pair_adv_manager_trigger_fmdn_provisioning, false);
+}
+
+static void fp_adv_fmdn_provisioning_trigger_manage(bool enable)
+{
+	if (enable) {
+		/* Fast Pair Implementation Guidelines for the locator tag use case:
+		 * after the Provider was paired, it should not change its MAC address
+		 * till FMDN is provisioned or till 5 minutes passes.
+		 */
+		(void) k_work_schedule(&fp_adv_rpa_suspension_work,
+				       K_MINUTES(FP_ADV_RPA_SUSPENSION_TIMEOUT));
+	} else {
+		(void) k_work_cancel_delayable(&fp_adv_rpa_suspension_work);
+	}
+
+	bt_fast_pair_adv_manager_request(
+		&bt_fast_pair_adv_manager_trigger_fmdn_provisioning, enable);
+}
+
+static void fp_adv_clock_sync_trigger_manage(bool enable)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_ADV_MANAGER_USE_CASE_LOCATOR_TAG_CLOCK_SYNC_TRIGGER)) {
+		return;
+	}
+
+	bt_fast_pair_adv_manager_request(&bt_fast_pair_adv_manager_trigger_clock_sync, enable);
+}
+
+static void fp_adv_clock_synced(void)
+{
+	LOG_DBG("Fast Pair Adv Manager: Locator Tag: clock information synchronized with "
+		"the authenticated Bluetooth peer");
+
+	if (fmdn_provisioned) {
+		/* Fast Pair Implementation Guidelines for the locator tag use case:
+		 * After a power loss, the device should advertise non-discoverable
+		 * Fast Pair frames until the next invocation of read beacon parameters.
+		 * This lets the Seeker detect the device and synchronize the clock even
+		 * if a significant clock drift occurred.
+		 */
+		fp_adv_clock_sync_trigger_manage(false);
+	}
+}
+
+static void fp_adv_provisioning_state_changed(bool provisioned)
+{
+	fmdn_provisioned = provisioned;
+
+	if (!bt_fast_pair_adv_manager_is_ready()) {
+		return;
+	}
+
+	if (provisioned) {
+		fp_adv_fmdn_provisioning_trigger_manage(false);
+	} else {
+		fp_adv_clock_sync_trigger_manage(false);
+	}
+}
+
+static struct bt_fast_pair_fmdn_info_cb fmdn_info_cb = {
+	.clock_synced = fp_adv_clock_synced,
+	.provisioning_state_changed = fp_adv_provisioning_state_changed,
+};
+
+static void fp_adv_account_key_written(struct bt_conn *conn)
+{
+	/* The first and only Account Key write starts the FMDN provisioning. */
+	if (!fmdn_provisioned && !fp_account_key_present &&
+	    bt_fast_pair_adv_manager_is_ready()) {
+		fp_adv_fmdn_provisioning_trigger_manage(true);
+	}
+
+	fp_account_key_present = bt_fast_pair_has_account_key();
+}
+
+static struct bt_fast_pair_info_cb fp_info_callbacks = {
+	.account_key_written = fp_adv_account_key_written,
+};
+
+static bool fp_adv_is_first_enable_after_bootup(void)
+{
+	static bool first_enable_after_bootup = true;
+	bool is_first_enable_after_bootup = first_enable_after_bootup;
+
+	first_enable_after_bootup = false;
+
+	return is_first_enable_after_bootup;
+}
+
+static int fp_adv_manager_locator_tag_enable_initial(void)
+{
+	/* Handle the clock synchronization trigger. */
+	if (fmdn_provisioned) {
+		fp_adv_clock_sync_trigger_manage(true);
+	}
+
```

Comment:
You forgot to add `fp_adv_clock_sync_trigger_manage(false);` when `!fmdn_provisioned`.

Reply:
Oh, actually this is not needed as the trigger could not be set before initial enable call anyway.